pair/src/instruction.rs

```diff
@@ -0,0 +1,157 @@
+use ft_logic_io::*;
+use ft_main_io::*;
+use gstd::{msg, prelude::*, ActorId};
+
+#[derive(Debug)]
+pub enum InstructionState {
+    ScheduledRun,
+    ScheduledAbort,
+    RunWithError,
+    Finished,
+}
+
+#[derive(Debug)]
+pub struct Instruction {
+    state: InstructionState,
+    address: ActorId,
+    transaction: FTokenAction,
+    compensation: Option<FTokenAction>,
+}
+
+impl Instruction {
+    pub fn new(
+        address: ActorId,
+        transaction: FTokenAction,
+        compensation: Option<FTokenAction>,
+    ) -> Self {
+        Instruction {
+            state: InstructionState::ScheduledRun,
+            address,
+            transaction,
+            compensation,
+        }
+    }
+
+    pub async fn start(&mut self) -> Result<(), ()> {
+        match self.state {
+            InstructionState::ScheduledRun => {
+                // Right now it's FTokenEvent, but should be moved
+                // LACKING CLONE OR COPY HERE :)
+                let result =
+                    msg::send_for_reply_as::<_, FTokenEvent>(self.address, self.transaction, 0)
```

Comment:
I'd prefer to avoid turbo-fish when possible as it is not too readable.
```suggestion
                let result: FTokenEvent =
                    msg::send_for_reply_as(self.address, self.transaction, 0)
```